tests/integrationv2/test_session_resumption.py

```diff
@@ -102,3 +100,127 @@ def test_session_resumption_s2n_client(managed_process, cipher, curve, protocol,
+
+    ticket_filename = 'session_ticket_' + port
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        insecure=True,
+        reconnect=False,
+        extra_flags = ['-sess_out', ticket_filename],
+        data_to_send = data_bytes(4096),
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.use_session_ticket = True
+    server_options.extra_flags = None
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+    # Client inputs stored session ticket to resume a session
+    assert os.path.exists(ticket_filename)
+    client_options.extra_flags = ['-sess_in', ticket_filename]
+
+    port = str(next(available_ports))
+    client_options.port = port
+    server_options.port = port
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    openssl_version = get_expected_openssl_version(protocol)
+    s2n_version = get_expected_s2n_version(protocol, OpenSSL)
+
+    # The client should have received a session ticket
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Post-Handshake New Session Ticket arrived:' in results.stdout
+        assert bytes("Protocol  : {}".format(openssl_version).encode('utf-8')) in results.stdout
+
+    # The server should indicate a session has been resumed
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Resumed session' in results.stdout
+        assert bytes("Actual protocol version: {}".format(s2n_version).encode('utf-8')) in results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_tls13_session_resumption_s2n_client(managed_process, cipher, curve, protocol, provider, certificate):
+    port = str(next(available_ports))
+
+    random_bytes = data_bytes(64)
+    num_full_connections = 1
+    num_resumed_connections = 5
```

Comment:
maybe add why this is `5`. this is the default value in s2nc

Reply:
if its not too difficult we should actually pass in `5` rather than using the default. that would avoid breakage if we ever update the default value in s2nc.

Reply:
I created an issue for this: https://github.com/aws/s2n-tls/issues/2838. Not sure if I'm going to make this change in the pr, but it certainly would be better than what we're currently doing.